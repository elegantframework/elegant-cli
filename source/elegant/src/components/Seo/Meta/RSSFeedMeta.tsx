interface Props {
    /**
     * The directory path of the rss feeds.
     */
    directory?: string;
    /**
     * Include the rss feed in the meta head?
     */
    enabled?: boolean;
};

/**
 * Create the rss feed head meta object data for the SEO component.
 * @returns A complete list of rss feeds.
 */
const RSSFeedMeta = ({
    directory = "/feeds",
    enabled = true,
}: Props) => {

    let data = [];

    if(enabled){
        data.push(
            {
                rel: 'alternate',
                type: 'application/rss+xml',
                title: 'RSS 2.0',
                href: `${directory}/feed.xml`
            },
            {
                rel: 'alternate',
                type: 'application/atom+xml',
                title: 'Atom 1.0',
                href: `${directory}/atom.xml`
            },
            {
                rel: 'alternate',
                type: 'application/json',
                title: 'JSON Feed',
                href: `${directory}/feeds.json`
            }
        );    
    };

    return data;
};

export default RSSFeedMeta;