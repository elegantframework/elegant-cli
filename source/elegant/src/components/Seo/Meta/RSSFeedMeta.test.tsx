import RSSFeedMeta from "./RSSFeedMeta";


describe('RSS Feed meta Component', () => {
    it('renders with data properly', () => {
        const result = RSSFeedMeta({
            directory: "",
        }); 

        expect(result).toEqual(
            expect.arrayContaining([
                expect.objectContaining({
                    rel: 'alternate',
                    type: 'application/rss+xml',
                    title: 'RSS 2.0',
                    href: `/feed.xml`
                })
            ])
        );

        expect(result).toEqual(
            expect.arrayContaining([
                expect.objectContaining({
                    rel: 'alternate',
                    type: 'application/atom+xml',
                    title: 'Atom 1.0',
                    href: `/atom.xml`
                })
            ])
        );

        expect(result).toEqual(
            expect.arrayContaining([
                expect.objectContaining({
                    rel: 'alternate',
                    type: 'application/json',
                    title: 'JSON Feed',
                    href: `/feeds.json`
                })
            ])
        );

        expect(result).toMatchSnapshot();
    });
});