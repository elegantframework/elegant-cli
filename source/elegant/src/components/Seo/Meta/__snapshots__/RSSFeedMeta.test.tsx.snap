// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RSS Feed meta Component renders with data properly 1`] = `
[
  {
    "href": "/feed.xml",
    "rel": "alternate",
    "title": "RSS 2.0",
    "type": "application/rss+xml",
  },
  {
    "href": "/atom.xml",
    "rel": "alternate",
    "title": "Atom 1.0",
    "type": "application/atom+xml",
  },
  {
    "href": "/feeds.json",
    "rel": "alternate",
    "title": "JSON Feed",
    "type": "application/json",
  },
]
`;
